### PR TITLE
Fix stacking order when making a tag floating

### LINKED ISSUE #1134: Stacking order not preserved when switching tag to floating
When switching the tag to floating, the stacking order is preserved which causes the focused window to be on the bottom of the stack and most likely be covered by other windows:
```
┌─(~)───────────────────────────────────────────────────────────
└──> herbstclient attr tags.focus.floating
false
┌─(~)───────────────────────────────────────────────────────────
└──> herbstclient stack                     
╾─╮
  ╰─╮ Monitor 0 with tag "four"
    ├─╼ Focus-Layer
    ├─╼ Fullscreen-Layer
    ├─╼ Floating-Layer
    ├─╮ Tiling-Layer
    │ ├─╼ Client 0x520000a "herbstclient stack (~) thorsten@hoth"
    │ ╰─╼ Client 0x4a0000a "unfocusedWindow"
    ╰─╮ Frame Layer
      ╰─╼ Window 0x80000c
┌─(~)───────────────────────────────────────────────────────────
└──> herbstclient attr tags.focus.floating on
┌─(~)───────────────────────────────────────────────────────────
└──> herbstclient stack                      
╾─╮
  ╰─╮ Monitor 0 with tag "four"
    ├─╼ Focus-Layer
    ├─╼ Fullscreen-Layer
    ├─╮ Floating-Layer
    │ ├─╼ Client 0x4a0000a "unfocusedWindow"
    │ ╰─╼ Client 0x520000a "herbstclient stack (~) thorsten@hoth"
    ├─╮ Tiling-Layer
    │ ├─╼ Client 0x520000a "herbstclient stack (~) thorsten@hoth"
    │ ╰─╼ Client 0x4a0000a "unfocusedWindow"
    ╰─╮ Frame Layer
      ╰─╼ Window 0x80000c
┌─(~)───────────────────────────────────────────────────────────
└──> 
```

### PR DESCRIPTION
**Pull request by t-wissmann**
When changing the floating state of a tag to 'on', use the existing
tiling window stacking order to ensure that the new stacking order in
the floating layer makes sense.

This fixes #1134.

### CODE REVIEW

File: tests/test_stack.py [hunk 1]
```diff
@@ -155,3 +155,32 @@ def test_stack_tree(hlwm):
...
+    layout = f"""
+    (split vertical:0.5:0
+        (clients max:0 {win[0]} {win[1]})
+        (clients max:0 {win[2]} {win[3]}))
+    """.replace('\n', '')
+    hlwm.call(['load', layout])
+    hlwm.call(['jumpto', win[focus_idx]])
+
+    stack_before = helper_get_stack_as_list(hlwm)
+    assert stack_before[0] == win[4]  # the floating must be on top anyway
```
Comment (blarz) [inline tests/test_stack.py:173]: floating window

File: tests/test_stack.py [hunk 2]
```diff
@@ -155,3 +155,32 @@ def test_stack_tree(hlwm):
...
+
+    hlwm.call('floating on')
+
+    # the stack should start with the floating window and the focused window
+    stack_expected = [win[4], win[focus_idx]]
+    # then the remaining tiled clients should follow and their order remains the same
+    stack_expected += [winid for winid in stack_before if winid not in stack_expected]
+    assert helper_get_stack_as_list(hlwm) == stack_expected
+
+    # turning floating of, just restores the old stack because
```
Comment (blarz) [inline tests/test_stack.py:183]: off

**Review by blarz — CHANGES_REQUESTED** [review]
Looks very good, tested locally too. Just two minor issues in the test

**Comment by t-wissmann** [comment]
Thanks! I've fixed both typos

**Review by blarz — APPROVED** [review]
Nice! Thanks!

**Pull request merged**